fcrepo-persistence-api/src/main/java/org/fcrepo/persistence/api/PersistentStorageSession.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.api;
+
+import java.time.Instant;
+
+import org.fcrepo.kernel.api.exception.CannotCreateResourceException;
+import org.fcrepo.kernel.api.exception.ItemNotFoundException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.models.Tombstone;
+
+/**
+ * An abstract class that mediates CRUD operations to and from persistence storage.
+ *
+ * @author dbernstein
+ * @author whikloj
+ */
+public interface PersistentStorageSession {
+
+    /**
+     * Return the transaction Id for this session, or null for a read-only session.
+     *
+     * @return the transaction id.
+     */
+    public String getTxId();
+
+    /**
+     * Create a new resource on the persistent storage.
+     *
+     * @param resource The new fedora resource to persist to storage.
+     * @return the resource with any new information related to the persisting..?
+     * @throws CannotCreateResourceException Error persisting the resource.
+     */
+    public FedoraResource create(final FedoraResource resource)
```

Comment:
Is the persistent storage session expected to produce new FedoraResource objects? Just wondering the value of returning FedoraResource in these methods. Alternatively, some of the methods could return PersistentStorageSession for method chaining.

Reply:
Unsure, that was just what we had documented. I guess the new resource could have additional properties due to being persisted that we would want?

Reply:
@whikloj : That was my understanding.

Reply:
@bbpennel ,  @whikloj :  maybe it _would_ make sense as @bbpennel suggested to return the session object on create() for chaining.  Same for update.